src/ConfigFile.c

```diff
@@ -678,6 +678,45 @@ void Ros_ConfigFile_ValidateNonCriticalSettings()
     }
 }
 
+void Ros_ConfigFile_PrintActiveConfiguration()
+{
+    Ros_Debug_BroadcastMsg("Config: ros_domain_id = %d", g_nodeConfigSettings.ros_domain_id);
+    Ros_Debug_BroadcastMsg("Config: node_name = %s", g_nodeConfigSettings.node_name);
+    Ros_Debug_BroadcastMsg("Config: node_namespace = %s", g_nodeConfigSettings.node_namespace);
+    Ros_Debug_BroadcastMsg("Config: remap_rules = %s", g_nodeConfigSettings.remap_rules);
+    Ros_Debug_BroadcastMsg("Config: agent_ip_address = %s", g_nodeConfigSettings.agent_ip_address);
+    Ros_Debug_BroadcastMsg("Config: agent_port_number = %s", g_nodeConfigSettings.agent_port_number);
+    Ros_Debug_BroadcastMsg("Config: sync_timeclock_with_agent = %d", g_nodeConfigSettings.sync_timeclock_with_agent);
+    Ros_Debug_BroadcastMsg("Config: namespace_tf = %d", g_nodeConfigSettings.namespace_tf);
+    Ros_Debug_BroadcastMsg("Config: publish_tf = %d", g_nodeConfigSettings.publish_tf);
+    Ros_Debug_BroadcastMsg("List of configured joint names:");
+    for (int i = 0; i < MAX_CONTROLLABLE_GROUPS; i += 1)
+    {
+        Ros_Debug_BroadcastMsg("---");
+        for (int j = 0; j < MP_GRP_AXES_NUM; j += 1)
+        {
+            if (strlen(g_nodeConfigSettings.joint_names[(i * MP_GRP_AXES_NUM) + j]) > 0)
+                Ros_Debug_BroadcastMsg(g_nodeConfigSettings.joint_names[(i * MP_GRP_AXES_NUM) + j]);
+            else
+                Ros_Debug_BroadcastMsg("x");
+        }
+    }
+    Ros_Debug_BroadcastMsg("---");
+    Ros_Debug_BroadcastMsg("Config: log_to_stdout = %d", g_nodeConfigSettings.log_to_stdout);
+    Ros_Debug_BroadcastMsg("Config: executor_sleep_period = %d", g_nodeConfigSettings.executor_sleep_period);
+    Ros_Debug_BroadcastMsg("Config: action_feedback_publisher_period = %d", g_nodeConfigSettings.action_feedback_publisher_period);
+    Ros_Debug_BroadcastMsg("Config: controller_status_monitor_period = %d", g_nodeConfigSettings.controller_status_monitor_period);
+    Ros_Debug_BroadcastMsg("Config: robot_status = %d", g_nodeConfigSettings.qos_robot_status);
+    Ros_Debug_BroadcastMsg("Config: joint_states = %d", g_nodeConfigSettings.qos_joint_states);
+    Ros_Debug_BroadcastMsg("Config: tf = %d", g_nodeConfigSettings.qos_tf);
+    Ros_Debug_BroadcastMsg("Config: tf_frame_prefix = %s", g_nodeConfigSettings.tf_frame_prefix);
+    Ros_Debug_BroadcastMsg("Config: stop_motion_on_disconnect = %d", g_nodeConfigSettings.stop_motion_on_disconnect);
+    Ros_Debug_BroadcastMsg("Config: inform_job_name = %s", g_nodeConfigSettings.inform_job_name);
+    Ros_Debug_BroadcastMsg("Config: allow_custom_inform_job = %d", g_nodeConfigSettings.allow_custom_inform_job);
+    Ros_Debug_BroadcastMsg("Config: userlan_monitor_enabled = %d", g_nodeConfigSettings.userlan_monitor_enabled);
+    Ros_Debug_BroadcastMsg("Config: userlan_monitor_port = %d", g_nodeConfigSettings.userlan_monitor_port);
```

Comment:
Just noticed this seems to print the active configuration, as opposed to the parsed one (including defaults).

My `.yaml` has:

```yaml
userlan_monitor_enabled: true
#userlan_monitor_port: USER_LAN1
```

MotoROS2 debug listener output:

```
[2023-07-14 14:42:04.815691] [192.168.255.1:64288]: FRI 2023-07-14 13:44:51.465 Config: userlan_monitor_enabled = 1
[2023-07-14 14:42:04.824265] [192.168.255.1:64288]: FRI 2023-07-14 13:44:51.465 Config: userlan_monitor_port = 1
```

This is sort-of correct, in that after auto-detection, `userlan_monitor_port` is set to `1`. But in the `.yaml`, it's not set (leading to auto-detection being activated).

I guess I should expect this to print the active configuration, given the name of the function (`Ros_ConfigFile_PrintActiveConfiguration()`). As a debug aid though, we might be interested in the parsed configuration.

Reply:
So this was more of a discussion than a request I guess.

I'm just not sure at the moment whether we'd want want over the other (ie: as-parsed or active).

What do you think?

Reply:
I think the user might prefer to know what the active configuration is if they are going through the log for troubleshooting purposes. They will have access to the configuration file and would know what value should be shown once it is parsed; whether it's set in the .yaml or not. 

One thing we can do is comment in mentioning this that auto-detect will be activated if the value is not set in the .yaml file. That way the user will have access to the active value and we could avoid the confusion of whether the value was parsed from the .yaml or not in the case where it is not set in the .yaml.

Reply:
> I think the user might prefer to know what the active configuration is if they are going through the log for troubleshooting purposes. 

Agreed.

> They will have access to the configuration file and would know what value should be shown once it is parsed; whether it's set in the .yaml or not.

well, that's the thing: do they? If the file is saved on the controller, they would not have (direct) access. Additionally, what's in the file and what the parser 'decides' is in the file are not necessarily the same thing.

> One thing we can do is comment in mentioning this that auto-detect will be activated if the value is not set in the .yaml file. That way the user will have access to the active value and we could avoid the confusion of whether the value was parsed from the .yaml or not in the case where it is not set in the .yaml.

I'm not sure I understand this part of your comment completely.

Reply:
> well, that's the thing: do they? If the file is saved on the controller, they would not have (direct) access.

I get that. Would it be excessive if we print out both the active and the parsed value? Seeing how both of them would be important for the user, depending upon whether the value is set in the .yaml or not and if it is, then what value is being parsed, it might be better to display both of the values. 

> I'm not sure I understand this part of your comment completely.

I meant to say that we can mention in the .yaml that auto-detect would be activated if there's no value set in the config file but all of that is already mentioned in the comments in the .yaml. Apologies for the oversight.

Reply:
Personally, I think printing both values would be excessive. I think printing the "active" settings is best, regardless of whether those settings were read from the config file or calculated through the code.

Reply:
The active configuration would indeed be the one that has the most influence on the runtime behaviour of MotoROS2. In that sense it would be the one to log -- anything not as it should be in there would be reason to start debugging.

I believe I'd agree with @ted-miller in that printing both might be too much.

Let's just keep the implementation as-is, but I'll make one small suggestion that would make things somewhat clearer.

Reply:
Thanks for the input @SejalBehere and @ted-miller.